### PR TITLE
Add statuses for posts impls

### PR DESCRIPTION
**Pull request by daria80**

### CODE REVIEW

File: src/main/java/com/reckue/post/transfers/nodes/image/ImageNodeResponse.java [hunk 1]
```diff
@@ -9,6 +9,7 @@
  *
  * @author Kamila Meshcheryakova
  */
+@EqualsAndHashCode(callSuper = true)
```
Comment (camelya58) [inline src/main/java/com/reckue/post/transfers/nodes/image/ImageNodeResponse.java:12]: I want to ask, why did you add "true" not "false"?

File: src/main/java/com/reckue/post/services/realizations/PostServiceRealization.java [hunk 2]
```diff
@@ -39,9 +40,30 @@ public Post create(Post post) {
         if (post.getNodes() != null) {
             post.getNodes().forEach(nodeService::create);
         }
+        validateOnCreateStatus(post);
         return postRepository.save(post);
     }
 
+    private void validateOnCreateStatus(Post post) {
+        if (post != null && post.getStatus() == PostStatusType.DRAFT) {
```
Comment (camelya58) [inline src/main/java/com/reckue/post/services/realizations/PostServiceRealization.java:48]: Don't understand the logic. Does it work?
I think that this method should create a status DRAFT for post, but it check existing status of post. But if the post has just been created, does the post have a status? i think it should be generated automatically.

File: src/main/java/com/reckue/post/transfers/nodes/image/ImageNodeResponse.java (hunk 1 above)
Comment (daria80) [reply src/main/java/com/reckue/post/transfers/nodes/image/ImageNodeResponse.java:12]: Because we override equals and hashcode for the superclass field as well

Comment (daria80) [reply src/main/java/com/reckue/post/transfers/nodes/image/ImageNodeResponse.java:12]: Because we override equals and hashcode for the superclass field as well

File: src/main/java/com/reckue/post/services/realizations/PostServiceRealization.java (hunk 2 above)
Comment (daria80) [reply src/main/java/com/reckue/post/services/realizations/PostServiceRealization.java:48]: When creating, we will always have a draft status

Comment (daria80) [reply src/main/java/com/reckue/post/services/realizations/PostServiceRealization.java:48]: When creating, we will always have a draft status
